tests/integration/questionnaire/test_questionnaire_endpoint_redirects.py

```diff
@@ -3,44 +3,48 @@
 from tests.integration.integration_test_case import IntegrationTestCase
 
 
-class TestQuestionnaireEndpointRedirects(IntegrationTestCase):
-    def test_get_invalid_questionnaire_location_redirects_to_latest(self):
+class TestQuestionnaireEndpoints(IntegrationTestCase):
+    BASE_URL = "/questionnaire"
+
+    def test_invalid_section_id_raises_404(self):
         # Given
-        self.launchSurvey("test_introduction")
+        self.launchSurvey("test_hub_and_spoke")
 
-        base_url = "/questionnaire/"
+        # When I navigate to the url for a section that does not exist
+        self.get(f"{self.BASE_URL}/sections/invalid-section/")
+
+        # Then I am shown a 404 page
+        self.assertStatusNotFound()
+
+    def test_get_invalid_questionnaire_location_raises_404(self):
+        # Given
+        self.launchSurvey("test_introduction")
 
         # When
-        self.get(base_url + "test")
+        self.get(f"{self.BASE_URL}/test")
 
         # Then
-        self.assertInUrl(base_url + "introduction")
+        self.assertStatusNotFound()
 
-    def test_post_invalid_questionnaire_location_redirects_to_latest(self):
+    def test_post_invalid_questionnaire_location_raises_404(self):
         # Given
         self.launchSurvey("test_introduction")
 
-        base_url = "/questionnaire/"
-
         # When
-        self.post(url=base_url + "test")
+        self.post(url=f"{self.BASE_URL}/test")
 
-        # Then
-        self.assertInUrl(base_url + "introduction")
+        # Then I am shown a 404 page
+        self.assertStatusNotFound()
 
-    def test_submit_answers_for_invalid_questionnaire_location_redirects_to_first_incomplete_location(
-        self
-    ):
+    def test_submit_answers_for_invalid_questionnaire_location_raises_404(self):
         # Given
         self.launchSurvey("test_textfield")
 
-        base_url = "/questionnaire/"
-
         # When
-        self.post(url=base_url + "min-max-block")
+        self.post(url=f"{self.BASE_URL}/min-max-block")
```

Comment:
Can't just go to base URL, as that will just redirect.

Reply:
Ok, but the test is named `test_submit_answers_for_invalid_questionnaire_location_raises_404` and if it's just testing an invalid block id, it is a duplicate of `test_post_invalid_questionnaire_location_raises_404`.

We should also consider raising an error for POST requests to `/questionnaire` when it's not available (rather than redirecting).

Reply:
Ideally this would raise a 404 on GET/POST, only don't because of how we do session launches at the moment. I have raised a card to look at that. For now, I will handle this for POST requests, so session launch still works as expected.

Reply:
Actually, I can't change this without changing how the hub continue button works. At the moment, a post on the route would redirect to the first incomplete location when questionnaire is not complete.